### DISCUSSION TITLE
Freeze cursor in position when drag area is triggered

### DISCUSSION THREAD

**Discussion by marrrc** [rxhanson/Rectangle#637] [Ideas]
hi,

how about an option to freeze the cursor in its current place whenever a drag area is activated, i.e. the footprint area appears? For me unfortunately it happens way too often that I e.g. want to have a window to be full-sized, so I drag it to the top edge of the screen, however I frequently move too much so that the macOS management area for multiple desktop appears. Could also be a modifier key, so whenever I move the mouse further, nothing else happens because the cursor is locked due to the footprint being activated.

**Comment by rxhanson** [comment]
This is a good idea. I’ll look into it for the next release.

**Reply by marrrc** [reply]
Thanks for looking into it! On further thought, it might not be as trivial, as a move "back" in the other direction of where the cursor came from (if one changes its mind about resizing the window) could be hindered. I then looked around, and noticed BetterTouchTool app has a similar functionality (https://imgur.com/lNjXcxu). It does not prevent my specific issue, but instead adds some sort of delay to the cursor movement - so if I drag the window a little further to the top edge than necessary (after footprint already activated), you can see that it kind of pushes the cursor back again for a brief while, so that it wouldn't leave the screen & activate mission control prematurely. Actually helps. Maybe this example can serve as an inspiration. :)
